### PR TITLE
JVM support Linux & Windows targets

### PR DESCRIPTION
**Pull request by nhachicha**
Add support for Linux (x86_64 Centos 7 and up) and Windows (x86_64 Wi…n 8.1 and up) for JVM targets

### CODE REVIEW

**Comment by nhachicha** [comment]
@cmelchior @rorbech CI is happy

File: Jenkinsfile [hunk 1]
```diff
@@ -96,6 +96,32 @@ pipeline {
                         runScm()
                     }
                 }
+
+                stage('build-jvm-native-libs') {
```
Comment (rorbech) [inline Jenkinsfile:101]: As already discussed we should maybe guard this with a flag, so it is easy to selectively build this only for publications and if you would like to verify it on a PR build.

File: Jenkinsfile [hunk 2]
```diff
@@ -509,3 +538,38 @@ def archiveServerLogs(String mongoDbRealmContainerId, String commandServerContai
...
+        sh """
+           cd packages/cinterop/src/jvmMain/linux/
+           rm -rf build-dir
+           mkdir build-dir
+           cd build-dir
+           cmake ..
+           make -j8
+        """
+
+        stash includes:'packages/cinterop/src/jvmMain/linux/build-dir/core/src/realm/object-store/c_api/librealm-ffi.so,packages/cinterop/src/jvmMain/linux/build-dir/librealmc.so', name: 'linux_so_files', allowEmpty: true
```
Comment (rorbech) [inline Jenkinsfile:559]: Does it make sense to `allowEmpty`?

File: Jenkinsfile [hunk 3]
```diff
@@ -509,3 +538,38 @@ def archiveServerLogs(String mongoDbRealmContainerId, String commandServerContai
 def runCommand(String command){
   return sh(script: command, returnStdout: true).trim()
 }
+
+def build_jvm_linux() {
+    unstash 'packages'
+    docker.build('jvm_linux', '-f packages/cinterop/src/jvmMain/linux/generic.Dockerfile .').inside {
+        sh """
+           cd packages/cinterop/src/jvmMain/linux/
+           rm -rf build-dir
```
Comment (rorbech) [inline Jenkinsfile:552]: We could consider allowing incremental builds for non-release builds. But guess it doesn't matter if we only build this for releases.

**Review by rorbech — APPROVED** [review]
LGTM. Could coalesce the cmake-files as we talked about and could consider parameterizing it for arch-types and release/debug, but not needed for this PR.

File: CHANGELOG.md [hunk 4]
```diff
@@ -14,7 +14,7 @@
 * Throw `Error` an unrecoverable Realm problem happen in the underlying storage engine.
 
 ### Enhancements
-* Add support for [JVM target](https://github.com/realm/realm-kotlin/issues/62) (currently only macos).
+* Add support for [JVM target](https://github.com/realm/realm-kotlin/issues/62).
```
Comment (cmelchior) [inline CHANGELOG.md:17]: Maybe mention that we don't build for M1's just yet?

File: Jenkinsfile [hunk 5]
```diff
@@ -509,3 +544,36 @@ def archiveServerLogs(String mongoDbRealmContainerId, String commandServerContai
...
+
+        stash includes:'packages/cinterop/src/jvmMain/linux/build-dir/core/src/realm/object-store/c_api/librealm-ffi.so,packages/cinterop/src/jvmMain/linux/build-dir/librealmc.so', name: 'linux_so_files', allowEmpty: true
+    }
+}
+
+def build_jvm_windows() {
+  def cmakeOptions = [
+        CMAKE_GENERATOR_PLATFORM: 'x64',
+        CMAKE_BUILD_TYPE: 'Release',
+        REALM_ENABLE_SYNC: "ON",
```
Comment (cmelchior) [inline Jenkinsfile:567]: Maybe add a comment here and a reference to an issue about combing CMake options?

File: packages/cinterop/src/jvmMain/linux/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,34 @@
+cmake_minimum_required(VERSION 3.4.1)
```
Comment (cmelchior) [inline packages/cinterop/src/jvmMain/linux/CMakeLists.txt:1]: Add a comment here about changes to this also needs to happen in the other files

File: packages/cinterop/src/jvmMain/windows/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,35 @@
+cmake_minimum_required(VERSION 3.4.1)
```
Comment (cmelchior) [inline packages/cinterop/src/jvmMain/windows/CMakeLists.txt:1]: Same, a comment here would be nice

**Review by cmelchior — APPROVED** [review]
Woop 🚀 . Only minor comments beside what Claus already mentioned.

File: Jenkinsfile (hunk 2 above)
Comment (nhachicha) [reply Jenkinsfile:559]: no, good catch

**Pull request merged**